docs/prod-host-setup.md

```diff
@@ -59,6 +59,20 @@ logging, which degraded the snapshot restore time from 3ms to 8.5ms on
 `aarch64`. In this case, creating the tap device for snapshot restore
 generated host kernel logs, which were very slow to write.
 
+### Logging and signal handlers
+
+Firecracker installs custom signal handlers for some of the POSIX signals, such
+as SIGSEGV, SIGSYS, etc.
+The custom signal handlers used by Firecracker are not async-signal-safe, since
+they write logs and flush the metrics, which use locks for synchronization.
+There is a very low probability, that, when running the signal handler on a
+thread, that thread was already holding a lock to the log or metrics buffers.
```

Comment:
This sentence is hard to read, I'll suggest this but feel free to rephrase however you see fit.

```diff
- There is a very low probability, that, when running the signal handler on a
- thread, that thread was already holding a lock to the log or metrics buffers.
+ It is possible but very unlikely that the handler will intercept a signal on a 
+ thread which is already holding a lock to the log or metrics buffer.
```